runtime/drivers/duckdb/transporter_duckDB_to_duckDB.go

```diff
@@ -90,16 +97,64 @@ func (t *duckDBToDuckDB) Transfer(ctx context.Context, srcProps, sinkProps map[s
 
 	// If the path is a local file reference, rewrite to a safe and repo-relative path.
 	if uri.Scheme == "" && uri.Host == "" {
-		sql, err := rewriteLocalPaths(ast, opts.RepoRoot, opts.AllowHostAccess)
+		rewrittenSQL, err := rewriteLocalPaths(ast, opts.RepoRoot, opts.AllowHostAccess)
 		if err != nil {
 			return fmt.Errorf("invalid local path: %w", err)
 		}
-		srcCfg.SQL = sql
+		srcCfg.SQL = rewrittenSQL
 	}
 
 	return t.to.CreateTableAsSelect(ctx, sinkCfg.Table, false, srcCfg.SQL)
 }
 
+func (t *duckDBToDuckDB) transferFromExternalDB(ctx context.Context, srcProps *dbSourceProperties, sinkProps *sinkProperties, opts *drivers.TransferOptions) error {
+	return t.to.WithConnection(ctx, 1, true, false, func(ctx, ensuredCtx context.Context, _ *sql.Conn) error {
+		res, err := t.to.Execute(ctx, &drivers.Statement{Query: "SELECT current_database(),current_schema();"})
+		if err != nil {
+			return err
+		}
+		defer res.Close()
+
+		res.Next()
+		var localDB, localSchema string
+		if err := res.Scan(&localDB, &localSchema); err != nil {
+			return err
+		}
```

Comment:
This seems dangerous – I think `t.to` will be locked until `res.Close()` has run, but since it's deferred, it won't run until the end of the func. Not exactly sure why this doesn't deadlock, but might be because there's some logic around `Next`/`Scan` auto-closing a `rows` if scanned to the end.

Would suggest the following changes:
- Consume the rows the usual way, i.e. using `for res.Nex()` – there will be just one result, so no harm in it.
- Explicitly call `res.Close` afterwards and in the error cases